### PR TITLE
First pass of visitors UI changes

### PR DESCRIPTION
**Pull request by damencho**
Screen Shot 2023-02-08 at 11 01 10 AM

Screen Shot 2023-02-08 at 1 01 39 PM

Screen Shot 2023-02-08 at 1 14 07 PM

image

### CODE REVIEW

**Comment by damencho** [comment]
@saghul can you take a look? 
This is only web, if we are ok with this approach for now I will work on the mobile.

**Comment by robertpin** [comment]
The yellow badge looks really weird to me. We already have a yellow badge for raised hands. Do we have a design for this?

File: react/features/base/icons/svg/index.ts
Comment (robertpin) [inline react/features/base/icons/svg/index.ts:90]: Please don't add back icons that we previously removed. Can't we use IconUser instead of adding IconUserGroups?

File: react/features/participants-pane/components/web/MeetingParticipants.tsx
Comment (robertpin) [inline react/features/participants-pane/components/web/MeetingParticipants.tsx:106]: This is a bad practice for `useSelector`. When you return the whole object from `useSelector` the component will re-render for any change in the object (not just the count). I know that now the object only has count and enabled but it's still a bad practice. Instead do
`const visitorsCount = useSelector((state: IReduxState) => state['features/visitors'].count);`

**Comment by saghul** [comment]
> The yellow badge looks really weird to me. We already have a yellow badge for raised hands. Do we have a design for this?

I do t think we do.

I think I'd add a "*" or "+" to the existing counter. Then when the participant pane opens the count will be revealed.

Generally speaking I don't think it's important to know the visitor count at a glance...

File: react/features/visitors/actionTypes.ts
Comment (robertpin) [inline react/features/visitors/actionTypes.ts:12]: Please update comment

File: react/features/visitors/actionTypes.ts
Comment (robertpin) [inline react/features/visitors/actionTypes.ts:15]: Please update comment

**Comment by robertpin** [comment]
> I do t think we do.

I see. I can say I'm also not a fan of how we have two titles in the participants pane now (Meeting participants and Visitors)

**Comment by robertpin** [comment]
@saghul I'm not entirely sure what this feature does so I can't approve it. I leave it to you :D

**Comment by damencho** [comment]
> The yellow badge looks really weird to me. We already have a yellow badge for raised hands. Do we have a design for this?

No. My intention is to be warning like ...

**Comment by damencho** [comment]
> I think I'd add a "*" or "+" to the existing counter. Then when the participant pane opens the count will be revealed.
> 
> Generally speaking I don't think it's important to know the visitor count at a glance...

@saghul What do you mean?

**Comment by saghul** [comment]
> > I think I'd add a "*" or "+" to the existing counter. Then when the participant pane opens the count will be revealed.
> > Generally speaking I don't think it's important to know the visitor count at a glance...
> 
> @saghul What do you mean?

In the current participant count we show just the number. My suggestion was to add some extra indicator after the number. I'll try to find some guidelines.

File: react/features/visitors/actionTypes.ts
Comment (damencho) [reply react/features/visitors/actionTypes.ts]: fixed

File: react/features/visitors/actionTypes.ts
Comment (damencho) [reply react/features/visitors/actionTypes.ts]: fixed

File: react/features/participants-pane/components/web/MeetingParticipants.tsx
Comment (damencho) [reply react/features/participants-pane/components/web/MeetingParticipants.tsx]: fixed

File: react/features/base/icons/svg/index.ts
Comment (damencho) [reply react/features/base/icons/svg/index.ts]: reverted this add and stick to the IconUsers.

**Comment by damencho** [comment]
I've added mobile impl and some small changes:
Screen Shot 2023-02-10 at 11 56 02 AM

Screen Shot 2023-02-14 at 8 17 34 AM

Screenshot_20230214-085328

@saghul @robertpin Can we do a second pass of this?

Also, there is a lint error that I'm not sure how to solve.

File: react/features/toolbox/components/native/Toolbox.js
Comment (saghul) [inline react/features/toolbox/components/native/Toolbox.js:103]: Can't visitors chat? Or not yet?

File: react/features/toolbox/components/native/Toolbox.js
Comment (saghul) [inline react/features/toolbox/components/native/Toolbox.js:118]: Shoudln't they be able to use the overflow button to get to settings to change their name for example?

**Review by saghul — COMMENTED** [review]
Left some comments, PTAL!

File: config.js
Comment (damencho) [reply config.js:49]: Well because of some other changes the visitors were broken because of the `//` vs `https://` setting. And that was why the first commit exists.  I can move it in another PR.

File: react/features/base/settings/functions.any.ts
Comment (damencho) [reply react/features/base/settings/functions.any.ts:124]: Yes exactly ... ok, I will change it to state[‘features/visitors’].iAmVisitor

File: react/features/toolbox/components/native/Toolbox.js
Comment (damencho) [reply react/features/toolbox/components/native/Toolbox.js:103]: Not yet.

File: react/features/toolbox/components/native/Toolbox.js
Comment (damencho) [reply react/features/toolbox/components/native/Toolbox.js:118]: Well ... there is no point as nobody will be seeing it.

File: conference.js
Comment (saghul) [reply conference.js:346]: Never mind I misread

File: config.js
Comment (saghul) [reply config.js:49]: Nah let's have it

File: react/features/toolbox/components/native/Toolbox.js
Comment (saghul) [reply react/features/toolbox/components/native/Toolbox.js:119]: There are many other options like enabling close captions for example. What about those?

"We can look into that later" is also a valid answer ;-)

Comment (damencho) [reply react/features/toolbox/components/native/Toolbox.js:119]: Yep. For sure there will be no cc for visitors for now ...

**Review by saghul — APPROVED** [review]

**Pull request merged**
